code/mission/missionparse.cpp

```diff
@@ -4590,6 +4594,12 @@ void parse_wing(mission *pm)
 				// assign wingnum
 				p_objp->wingnum = wingnum;
 				p_objp->pos_in_wing = i;
+
+				// we have found our "special ship" (our wing leader)
+				if (!Fred_running && i == 0){
+					wingp->special_ship_ship_info_index = p_objp->ship_class;
+				}
+
```

Comment:
Don't initialize the wing field here, because this is when ships are parsed, not created.  Do it in `parse_create_object_sub`, probably in the block commented with `// if ship is in a wing, set some equivalent flags on the ship`.  Since that function is called when any ship in the wing is created, it will set the variable in any circumstance, even if there are scripting shenanigans.

Reply:
All right